### PR TITLE
Terraform Driver examples for accessing private Git and using secret refs

### PR DESCRIPTION
**Pull request by TobiasBabin**
This PR adds a number of examples on the Terraform Driver showing

- the use of secret references
- how to access a private Git repository
- how to use the `azure-blob` and `s3` Resource Types

### CODE REVIEW

File: resource-definitions/terraform-driver/private-git-repo/ssh-secret-refs.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+      variables = {
+        # ...
+      }
+    })
+
+    values_string = jsonencode({
+      # Connection information to the target Git repo
+      source = {
+        path = var.tf_module_github_path
```
Comment (mathieu-benoit) [inline resource-definitions/terraform-driver/private-git-repo/ssh-secret-refs.tf:28]: To be consistent with the `s3` example in this PR, and to be more explicit for our end users, let's change these `var`s by:
```
          path = "s3/terraform/bucket/"
          rev  = "refs/heads/main"
          url  = "https://my-domain.com/my-org/my-repo.git"
```

With appropriate `url` if ssh.

In all the different places in this PR too.

Thanks!

**Review by mathieu-benoit — CHANGES_REQUESTED** [review]
minor feedback for consistency, otherwise LGTM

File: resource-definitions/terraform-driver/private-git-repo/ssh-secret-refs.tf (hunk 1 above)
Comment (TobiasBabin) [reply resource-definitions/terraform-driver/private-git-repo/ssh-secret-refs.tf]: Makes sense, thanks! I pushed changes to that end.

**Review by mathieu-benoit — APPROVED** [review]
LGTM

**Pull request merged**